plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java

```diff
@@ -335,6 +335,97 @@ public void testPredicatePushdown(String value, String operator)
         }
     }
 
+    @Test(dataProvider = "complexPredicatePushdownProvider")
+    public void testComplexExpressionPredicatePushdown(String value, String operator)
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_complex_expression_predicate_pushdown", "AS SELECT %1$s col, %1$s another_col".formatted(value))) {
+            assertThat(query("SELECT * FROM %1$s WHERE col %2$s %3$s OR another_col %2$s %3$s".formatted(table.getName(), operator, value)))
+                    .isFullyPushedDown();
+
+            assertThat(query("SELECT * FROM %1$s WHERE col %2$s %3$s OR another_col %2$s NULL".formatted(table.getName(), operator, value)))
+                    .isFullyPushedDown();
+
+            assertThat(query("SELECT * FROM %s WHERE col %s %s OR another_col IS NULL".formatted(table.getName(), operator, value)))
+                    .isFullyPushedDown();
+
+            assertThat(query("SELECT * FROM %s WHERE col %s %s OR another_col IS NOT NULL".formatted(table.getName(), operator, value)))
+                    .isFullyPushedDown();
+
+            assertThat(query("SELECT * FROM %1$s WHERE col %2$s %3$s OR another_col IN (NULL, %3$s)".formatted(table.getName(), operator, value)))
+                    .isFullyPushedDown();
+
+            assertThat(query("SELECT * FROM %1$s WHERE col %2$s %3$s OR another_col NOT IN (NULL, %3$s)".formatted(table.getName(), operator, value)))
+                    .isFullyPushedDown();
+        }
```

Comment:
Can we have some sort of nested expression evaluation ? In case of numeric constant with negative numbers ?